File: app/src/main/java/net/mieczkowski/yelpbusinessexample/recyclerAdapters/searchHistory/SearchHistoryViewHolder.java
package net.mieczkowski.yelpbusinessexample.recyclerAdapters.searchHistory;

import android.view.View;
import android.widget.TextView;

import net.mieczkowski.yelpbusinessexample.R;
import net.mieczkowski.yelpbusinessexample.models.PreviousSearch;
import net.mieczkowski.yelpbusinessexample.recyclerAdapters.base.BaseViewHolder;

import butterknife.BindView;

/**
 * Created by Josh Mieczkowski on 10/20/2017.
 */

public class SearchHistoryViewHolder extends BaseViewHolder<PreviousSearch> {

    @BindView(R.id.txtSearchHistory)
    TextView txtSearchHistory;

    public SearchHistoryViewHolder(View itemView) {
        super(itemView);
    }

    @Override
    public void onBind(PreviousSearch object) {
        txtSearchHistory.setText(object.getSearchTerm());
    }
}

File: app/src/main/java/net/mieczkowski/yelpbusinessexample/recyclerAdapters/searchHistory/SearchHistoryViewHolder.kt
package net.mieczkowski.yelpbusinessexample.recyclerAdapters.searchHistory

import android.view.View
import android.widget.TextView
import kotlinx.android.synthetic.main.row_search_history.view.*
import net.mieczkowski.dal.cache.models.PreviousSearch
import net.mieczkowski.yelpbusinessexample.recyclerAdapters.base.BaseViewHolder

/**
 * Created by Josh Mieczkowski on 10/20/2017.
 */

class SearchHistoryViewHolder(itemView: View) : BaseViewHolder<PreviousSearch>(itemView) {

    val txtSearchHistory: TextView = itemView.txtSearchHistory

    override fun onBind(item: PreviousSearch?) {
        txtSearchHistory.text = item?.searchTerm
    }
}
